core/src/main/scala/org/http4s/headers/Forwarded.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2013-2020 http4s.org
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.http4s.headers
+
+import java.net.{Inet4Address, Inet6Address}
+
+import cats.data.NonEmptyList
+import org.http4s.parser.{ForwardedModelParsers, HttpHeaderParser}
+import org.http4s.util.Writer
+import org.http4s.{Header, HeaderKey, ParseResult, Uri}
+
+object Forwarded
+    extends HeaderKey.Internal[Forwarded]
+    with HeaderKey.Recurring
+    with ForwardedModelParsers {
+
+  final case class Node(nodeName: Node.Name, nodePort: Option[Node.Port] = None)
+
+  object Node {
+    def apply(nodeName: Name, nodePort: Port): Node = apply(nodeName, Some(nodePort))
+
+    sealed trait Name extends Any
```

Comment:
I've never been good at predicting `AnyVal` boxing, but since it's polymorphic in `Node`, won't these always be boxed in practice?  I'm not sure the `Any` / `AnyRef` is buying anything here, but I could be wrong.

Reply:
I added these at the very beginning of my work, just in case, but I agree – there's no real sense to keep them.